src/ir/ty.rs

```diff
@@ -835,6 +835,22 @@ impl Type {
             _ => {}
         }
 
+        // Objective C template type parameter
+        if ty_kind == CXType_Typedef
```

Comment:
Also, what guarantees that every template parameter appears as a `Typedef`? I don't see anything that would imply that, care to explain?